app/api/auth/create_access_token.go

```diff
@@ -21,26 +23,37 @@ import (
 	"github.com/France-ioi/AlgoreaBackend/app/service"
 )
 
+type ctxKey int
+
+const parsedRequestData ctxKey = iota
+
 // swagger:operation POST /auth/token auth accessTokenCreate
 // ---
 // summary: Create or refresh an access token
 // description:
-//     If the "Authorization" header is not given, the service converts the given OAuth2 authorization code into tokens,
+//     If none of the "Authorization" header and "access_token" cookie are given,
+//     the service converts the given OAuth2 authorization code into tokens,
 //     creates or updates the authenticated user in the DB with the data returned by the login module,
 //     and saves new access & refresh tokens into the DB as well.
 //     If OAuth2 authentication has used the PKCE extension, the `{code_verifier}` should be provided
 //     so it can be sent together with the `{code}` to the authentication server.
 //
 //
-//     If the "Authorization" header is given, the service refreshes the access token
+//     If the "Authorization" header or/and the "access_token" is given
+//     (when both are given, the "Authorization" header is used),
+//     the service refreshes the access token
 //     (locally for temporary users or via the login module for normal users) and
 //     saves it into the DB keeping only the input token (from authorization headers) and the new token.
 //     Since the login module responds with both access and refresh tokens, the service updates the user's
 //     refresh token in this case as well. If there is no refresh token for the user in the DB,
 //     the 'not found' error is returned.
 //
 //
-//   * One of the “Authorization” header and the `{code}` parameter should be present (not both at once).
+//   * The "Authorization" header is not allowed when the `{code}` is given.
+//
+//   * The "access_token" cookie is not allowed when the `{code}` is given and `{use_cookie}`=1.
```

Comment:
The client does not control the "access_token" cookie. If there is a dummy "access_token" cookie, the user cannot re-login with that rule and so is stuck forever?
Instead, the cookie could be ignored, and it would probably be overridden by the new one.

Reply:
fixed